trie/trie.go

```diff
@@ -287,34 +380,46 @@ func concat(s1 []byte, s2 ...byte) []byte {
 	return r
 }
 
-func (t *Trie) resolve(n node) node {
+func (t *Trie) resolve(n node, prefix, suffix []byte) (node, error) {
 	if n, ok := n.(hashNode); ok {
-		return t.resolveHash(n)
+		return t.resolveHash(n, prefix, suffix)
+	}
+	return n, nil
+}
+
+type MissingNodeError struct {
+	RootHash, NodeHash   common.Hash
+	KeyPrefix, KeySuffix []byte
+}
+
+func (err *MissingNodeError) Error() string {
+	return fmt.Sprintf("Missing trie node %064x", err.NodeHash)
+}
+
+func unhandledError(err error) {
+	if glog.V(logger.Error) {
+		glog.Errorf("Unhandled trie error: %v", err)
```

Comment:
I don't think it's a good idea to put a log statement in a helper function. The reason is that logs are tagged with line numbers that should provide a hint as to where the issue went wrong. With this approach all error logs will have the same line number, loosing part of its information content.

Further, you can simplify the error reporting to:

```
glog.V(logger.Error).Errorf("Unhandled trie error: %v", err)
```